js/WebViewTypes.js

```diff
@@ -0,0 +1,332 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ * @flow
+ */
+
+'use strict';
+
+import type {Node, Element} from 'react';
+import type {EdgeInsetsProp} from 'EdgeInsetsPropType';
+import type {ViewStyleProp} from 'StyleSheet';
+import type {ViewProps} from 'ViewPropTypes';
+
+export type WebViewErrorEvent = {
+  domain: any,
+  code: any,
+  description: any,
+};
+
+export type WebViewEvent = Object;
+
+export type DataDetectorTypes =
+  | 'phoneNumber'
+  | 'link'
+  | 'address'
+  | 'calendarEvent'
+  | 'trackingNumber'
+  | 'flightNumber'
+  | 'lookupSuggestion'
+  | 'none'
+  | 'all';
+
+export type WebViewSourceUri = {|
+  /**
+   * The URI to load in the `WebView`. Can be a local or remote file.
+   */
+  uri?: ?string,
+
+  /**
+   * The HTTP Method to use. Defaults to GET if not specified.
+   * NOTE: On Android, only GET and POST are supported.
+   */
+  method?: string,
+
+  /**
+   * Additional HTTP headers to send with the request.
+   * NOTE: On Android, this can only be used with GET requests.
+   */
+  headers?: Object,
+
+  /**
+   * The HTTP body to send with the request. This must be a valid
+   * UTF-8 string, and will be sent exactly as specified, with no
+   * additional encoding (e.g. URL-escaping or base64) applied.
+   * NOTE: On Android, this can only be used with POST requests.
+   */
+  body?: string,
+|};
+
+export type WebViewSourceHtml = {|
+  /**
+   * A static HTML page to display in the WebView.
+   */
+  html?: ?string,
+  /**
+   * The base URL to be used for any relative links in the HTML.
+   */
+  baseUrl?: ?string,
+|};
+
+export type WebViewSource = WebViewSourceUri | WebViewSourceHtml;
+
+export type WebViewNativeConfig = $ReadOnly<{|
+  /*
+   * The native component used to render the WebView.
+   */
+  component?: ?any,
```

Comment:
A component that is passed in appears that it must also take the same props as the existing WebView, so this should probably be something like
```
React.ComponentType<WebViewSharedProps>
```